src/Store.cs

```diff
@@ -340,41 +340,51 @@ private static class Native
 
         private static readonly Native.Finalizer Finalizer = (p) => GCHandle.FromIntPtr(p).Free();
 
-        private readonly ConcurrentDictionary<ExternFunc, Function> _externFuncCache = new();
-        private readonly ConcurrentDictionary<ExternMemory, Memory> _externMemoryCache = new();
-        private readonly ConcurrentDictionary<ExternGlobal, Global> _externGlobalCache = new();
+        private readonly ConcurrentDictionary<(ExternKind kind, ulong store, nuint __private), object> _externCache = new();
 
         internal Function GetCachedExtern(ExternFunc @extern)
         {
-            if (!_externFuncCache.TryGetValue(@extern, out var func))
+            // We use a `ValueTuple` as key, consisting of the extern type and the both
+            // struct fields, which works since all `Extern...` structs have the same
+            // fields.
+            // Even though the second field is named "__private", it should be Ok to
+            // access it since we won't interpret the value in any way, but just use it
```

Comment:
The `__private` field is declared in struct definitions (like `wasmtime_func `) the [`extern.h`](https://github.com/bytecodealliance/wasmtime/blob/main/crates/c-api/include/wasmtime/extern.h) header file. However I'm nut sore where to find its actual declaration, as I'm not a Rust developer (the `extern.rs` seems to import types like `wasmtime::Func` etc. from some other files).

Can you elaborate more on your question? Do you mean whether it could happen that e.g. `wasmtime_instance_export_get` returns a different struct value (e.g. `wasmtime_func`) for a specific extern name over the lifetime of the instance? I don't think that can be the case; even if it would be the case, it shouldn't be much of a problem because in that case we would create a new entry in the dictionary and return a new instance (e.g. `Function`).

Reply:
>  don't think that can be the case; even if it would be the case, it shouldn't be much of a problem because in that case we would create a new entry in the dictionary and return a new instance (e.g. Function).

Yes this was the gist of the question.  That's what I assumed but wanted to double check.  Thanks!